src/mem/epoch/markable_atomic.rs

```diff
@@ -5,33 +5,38 @@ use std::sync::atomic::{self, Ordering};
 
 use super::{Owned, Shared, Guard};
 
-/// Like `std::sync::atomic::AtomicPtr`.
+/// Provides atomic access to a `(*const T, usize)` pair.
 ///
-/// Provides atomic access to a (nullable) pointer of type `T`, interfacing with
-/// the `Owned` and `Shared` types.
+/// The pair consists of:
+///
+/// - a (nullable) pointer of type `T`, interfacing with the `Owned` and `Shared` types
+///
+/// - a `usize` marker value that can fit into the unused bits of the pointer:
+/// `mark < std::mem::align_of::<T>()`
 #[derive(Debug)]
-pub struct MarkedAtomic<T> {
+pub struct MarkableAtomic<T> {
```

Comment:
This seems like a good example of [Wadler's Law](https://wiki.haskell.org/Wadler's_Law) :) Fair, i'll change it to [`TaggedAtomic`](https://en.wikipedia.org/wiki/Tagged_pointer)